Upstream: yes
From 97738597f6fd2b98127bca398d520c81f09be213 Mon Sep 17 00:00:00 2001
From: Michal Sekletar <msekletar@users.noreply.github.com>
Date: Fri, 26 Aug 2016 00:07:58 +0200
Subject: [PATCH] units: remove udev control socket when systemd stops the
 socket unit (#4039)

Mere presence of the socket in the filesystem makes
udev_queue_get_udev_is_active() return that udev is running. Note that,
udev on exit doesn't unlink control socket nor does systemd. Thus socket
stays around even when both daemon and socket are stopped. This causes
problems for cryptsetup because when it detects running udev it launches
synchronous operations that *really* require udev. This in turn may
cause blocking and subsequent timeout in systemd-cryptsetup on reboot
while machine is in a state that udev and its control socket units are
stopped, e.g. emergency mode.

Fixes #2477
(cherry picked from commit 51bce29f8eee1cdc0eb25c8fe87e69e24b5fbd98)
---
 units/systemd-udevd-control.socket | 1 +
 1 file changed, 1 insertion(+)

diff --git a/units/systemd-udevd-control.socket b/units/systemd-udevd-control.socket
index 8330a1c..46f704e 100644
--- a/units/systemd-udevd-control.socket
+++ b/units/systemd-udevd-control.socket
@@ -17,3 +17,4 @@ Service=systemd-udevd.service
 ListenSequentialPacket=/run/udev/control
 SocketMode=0600
 PassCredentials=yes
+RemoveOnStop=yes
